File: MonityApp/Models/Category.swift
import Foundation

struct Category: Codable, Identifiable {
    let id: Int
    var name: String
    var nameHe: String?
    var icon: String
    var color: String
    var type: CategoryType
    var isDefault: Bool

    enum CategoryType: String, Codable {
        case expense
        case income
        case both
    }

    var localizedName: String {
        let lang = UserDefaults.standard.string(forKey: "app_language") ?? "he"
        if lang == "he", let he = nameHe, !he.isEmpty {
            return he
        }
        return name
    }
}

struct CategoryListResponse: Codable {
    let categories: [Category]
}

struct CategorySingleResponse: Codable {
    let category: Category
}

File: backend/src/models/Category.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Category = sequelize.define('Category', {
  id: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true,
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  nameHe: {
    type: DataTypes.STRING,
    defaultValue: '',
  },
  icon: {
    type: DataTypes.STRING,
    defaultValue: '💰',
  },
  color: {
    type: DataTypes.STRING(7),
    defaultValue: '#007AFF',
  },
  type: {
    type: DataTypes.ENUM('expense', 'income', 'both'),
    defaultValue: 'expense',
  },
  isDefault: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
  },
});

module.exports = Category;
